dask_expr/collection.py

```diff
@@ -118,29 +129,32 @@ def __getitem__(self, other):
 
     def __dask_graph__(self):
         out = self.expr
-        out = out.simplify()
+        out = out.simplify().lower()
         return out.__dask_graph__()
 
     def __dask_keys__(self):
         out = self.expr
-        out = out.simplify()
+        out = out.simplify().lower()
         return out.__dask_keys__()
 
     def simplify(self):
         return new_collection(self.expr.simplify())
 
+    def lower(self):
+        return new_collection(self.expr.lower())
+
     @property
     def dask(self):
         return self.__dask_graph__()
 
     def __dask_postcompute__(self):
-        state = self.simplify()
+        state = self.simplify().lower()
         if type(self) != type(state):
             return state.__dask_postcompute__()
         return _concat, ()
 
     def __dask_postpersist__(self):
-        state = self.simplify()
+        state = self.simplify().lower()
```

Comment:
I think there was some conversation around simplifying after lowering.  Maybe we want to replace these with `.optimize()` which then can choose to alternate between the two if it wants?

This gets a little weird because of fusion, but maybe we start controlling that with config instead of a keyword.

Reply:
We had some discussion about this in #194 and agreed that a more general mechanism probably makes more sense. Adding lower felt not extensible enough (starting with the simplify -> lower -> simplify things)